src/Engine/Combat/BattleInstance.java

```diff
@@ -59,42 +62,85 @@ public BattleInstance(Unit pAttacker, Unit pDefender, GameMap map, int attackerX
    */
   public BattleSummary calculateBattleResults()
   {
+    CombatContext context = new CombatContext(gameMap, attacker, attackerWeapon, defender, defenderWeapon, battleRange, attackerX, attackerY);
+    Map<Unit, Entry<Weapon,Double>> unitDamageMap = new HashMap<Unit, Entry<Weapon,Double>>();
+    unitDamageMap.put(attacker, new AbstractMap.SimpleEntry<Weapon,Double>(attackerWeapon, 0.0));
+    unitDamageMap.put(defender, new AbstractMap.SimpleEntry<Weapon,Double>(attackerWeapon, 0.0));
+    
     // let the COs fool around with anything they want...
-    attacker.CO.changeCombatContext(this);
-    defender.CO.changeCombatContext(this);
+    attacker.CO.changeCombatContext(context);
+    defender.CO.changeCombatContext(context);
+    
+    // From here on in, use context variables only (Easy way to confirm: delete BattleInstance's variable block and see what gets marked as an error)
     
     double attackerHPLoss = 0;
 
     // Set up our scenario.
-    BattleParams attackInstance = new BattleParams(attacker, attackerWeapon,
-        defender, gameMap.getEnvironment(defenderX, defenderY), false, this);
+    BattleParams attackInstance = new BattleParams(context, false);
 
     // Last-minute adjustments.
-    attacker.CO.applyCombatModifiers(attackInstance, true);
-    defender.CO.applyCombatModifiers(attackInstance, false);
+    context.attacker.CO.applyCombatModifiers(attackInstance, true);
+    context.defender.CO.applyCombatModifiers(attackInstance, false);
 
     double defenderHPLoss = attackInstance.calculateDamage();
-    if( !isSim && defenderHPLoss > defender.getPreciseHP() ) defenderHPLoss = defender.getPreciseHP();
+    unitDamageMap.put(context.attacker, new AbstractMap.SimpleEntry<Weapon,Double>(context.attackerWeapon, defenderHPLoss));
+    if( !isSim && defenderHPLoss > context.defender.getPreciseHP() ) defenderHPLoss = context.defender.getPreciseHP();
 
     // If the unit can counter, and wasn't killed in the initial volley, calculate return damage.
-    if( canCounter && (defender.getPreciseHP() > defenderHPLoss) )
+    if( context.canCounter && (context.defender.getPreciseHP() > defenderHPLoss) )
     {
       // New battle instance with defender counter-attacking.
-      BattleParams defendInstance = new BattleParams(defender, defenderWeapon,
-          attacker, gameMap.getEnvironment(attackerX, attackerY), true, this);
+      BattleParams defendInstance = new BattleParams(context, true);
       defendInstance.attackerHP -= defenderHPLoss; // Account for the first attack's damage to the now-attacker.
 
       // Modifications apply "attacker first", and the defender is now the attacker.
-      defender.CO.applyCombatModifiers(attackInstance, true);
-      attacker.CO.applyCombatModifiers(attackInstance, false);
+      context.defender.CO.applyCombatModifiers(attackInstance, true);
+      context.attacker.CO.applyCombatModifiers(attackInstance, false);
```

Comment:
Shouldn't these two be `defendInstance`?

Reply:
This kind of thing (or maybe, code that uses these hooks) should really have test cases to prove they do what we expect, especially since nobody's been using it until now. And yes, I know we don't have a lot of Commander test cases, but that's partly because they've been fairly simple until now, and not mucking about with game mechanics too much.

Reply:
....you are absolutely correct.

As for testing, I agree with CombatEngine/BattleInstance tests, and probably bringing in Venge (Cinder too?) to mess with stuff.

I guess I should work on that, then.